plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -366,7 +368,33 @@ public synchronized void createDatabase(HiveIdentity identity, Database database
 
     public synchronized void dropDatabase(HiveIdentity identity, String schemaName)
     {
-        setExclusive((delegate, hdfsEnvironment) -> delegate.dropDatabase(identity, schemaName, true));
+        HdfsContext context = new HdfsContext(
+                identity.getUsername()
+                        .map(ConnectorIdentity::ofUser)
+                        .orElseThrow(() -> new IllegalStateException("username is null")));
```

Comment:
Use `new HdfsContext(session)`. session or ConnectorIdentity may care information such as extra credentials that's used to access file system (this is the very use-case for which extra credentials were originally added)